.github/workflows/python_tests.yml

```diff
@@ -48,8 +48,6 @@ jobs:
         run: |
           git fetch --no-tags --prune --depth=1 origin master
           make test-py
-          source scripts/run_doctests.sh
```

Comment:
@hornc, I brought this up at the weekly ABC call and as Drini mentioned removing `mypy` is okay here, but we'd like to keep running the doctests.

Reply:
@scottbarnes the majority of the tests run in `source scripts/run_doctests.sh` are _already_ running in `make test-py` 

Running them twice does not add value, it simply takes longer and gives a false impression of 'doing more testing'. It also makes it harder to debug and locate when a test fails if it is failing in two locations.

In general, I don't think the OL project's doctests are maintained or current.  The majority of the tests run in `/run_doctests.sh` are _not_ doctests anyway.

The correct place to put OL Python tests are in the standard test files to be run via `make test-py`

I believe that this has already been done, but it's hard to review since the bulk of what runs  is simple duplication.

The ignore list in `scripts/run_doctests.sh` seems totally arbitrary. This script should be deprecated, and if there is anything still of value in doctests, they should be highlighted and moved to more appropriate locations, so they can be maintained properly.

Reply:
Ah, thank you for pointing out that the they're already running from `make test-py`. I will try to look at this more closely tomorrow.

Reply:
@hornc, I created #9929 to address this. We will discuss this during milestone planning, and I will mention that I think that issue is blocking this PR, which is itself blocking another PR, if I understand correctly.

Reply:
I'm excited for this PR to land, the mypy changes and test + workflow changes shouldn't be required for it to land. Making the frustrating call to separate these two issues (as @scottbarnes has already been doing). The changes in this file should be reverted, we can test with `make  test-py` inside the container. If there's a separate PR we need to address test cleanup, we can talk during community call when all stakeholders are available and all are welcome to join.

Reply:
@mekarpeles @scottbarnes I've reinstated the doctests, the overall tests still pass, which is as expected. This should now be ready for final review and merge.

As stated above, there is a follow-up PR to fill out the contributor role abbreviations as a stand-alone feature / improvement:  #9901